### ISSUE TITLE
Hide selected "My blocks"

### ISSUE THREAD

**Issue by piw-piw** [ScratchAddons/ScratchAddons#4548]
Makes invisible custom blocks
```
sa-inside_blocks:{"_encode %s"}
```
![image](https://user-images.githubusercontent.com/76477070/167304267-4dd6fedc-82d5-4f14-bda6-c0fa95ba47ac.png)

**Comment by WorldLanguages** [comment]
@piw-piw Can you elaborate?

**Comment by Chiroyce1** [comment]
@WorldLanguages in some programming languages (mainly python but maybe also in others) adding a `_` at the beginning of the method (function) name makes it a function that's not meant to be used directly, but it can be used internally by a program or another function. 

So if you have a function that calculates the area of a circle in a given image, you might want to put the function that finds it's radius in such type of a function, so this way the user can only use the area function. 

[Here's](https://peps.python.org/pep-0008/#descriptive-naming-styles) some more info (search for `_single_leading_underscore`).

**Comment by CST1229** [comment]
> This does raise some concern, since if a block is made hidden, how can it be brought onto the code area?

Checkbox to show hidden blocks as transparent at the top of the My Blocks category.

**Comment by piw-piw** [comment]
> Checkbox to show hidden blocks as transparent at the top of the My Blocks category.

Or if it's possible, make list to expand.
![image](https://user-images.githubusercontent.com/76477070/167480229-61d6afeb-556e-463d-98b5-02448ffb7145.png)

**Comment by piw-piw** [comment]
> @WorldLanguages in some programming languages (mainly python but maybe also in others) adding a `_` at the beginning of the method (function) name makes it a function that's not meant to be used directly, but it can be used internally by a program or another function.

It isn't designated by `_`, but by `sa-inside_blocks:{"_encode %s"}` comment.

**Comment by RedGuy12** [comment]
We'd probably use _, we don't want to use a sa- prefix


________________________________

**Comment by WorldLanguages** [comment]
Huh, I see.

@piw-piw How many custom blocks do your projects often have?

**Comment by piw-piw** [comment]
If I'm working with framework - 20, if not 5.

**Comment by Cool-Scratcher** [comment]
How would this be useful? I don't see any use case for this, but maybe that's just me being me lol

**Comment by mxmou** [comment]
> How would this be useful? I don't see any use case for this, but maybe that's just me being me lol

It's for custom blocks that are only used to implement other custom blocks.

**Comment by lisa-wolfgang** [comment]
I've been thinking about a similar feature for variables and lists.

To solve the block access issue, the block palette entry could be relocated to the custom block it's used in. It would go in the same area as custom block parameters:

![image](https://user-images.githubusercontent.com/43426138/169946831-291c5512-9493-466b-9ed2-13cd4b8b796c.png)

**Comment by mxmou** [comment]
> To solve the block access issue, the block palette entry could be relocated to the custom block it's used in. It would go in the same area as custom block parameters:

Does this mean the user would have to know where the block will be used when creating it?

**Comment by piw-piw** [comment]
> I've been thinking about a similar feature for variables and lists.

Yes, it's also nice feature.

**Comment by lisa-wolfgang** [comment]
> Does this mean the user would have to know where the block will be used when creating it?

No. The thought was that the editor would manage this automatically, only hiding the block from the palette and showing the block in the definition once that custom block is only used in that one definition.

**Comment by mxmou** [comment]
> > Does this mean the user would have to know where the block will be used when creating it?
> 
> No. The thought was that the editor would manage this automatically, only hiding the block from the palette and showing the block in the definition once that custom block is only used in that one definition.

Sometimes "internal" blocks are used by more than one other block. It's also possible that the user wants to use the block in other places later.
